skimage/__init__.py

```diff
@@ -56,140 +56,7 @@
-        import warnings
-        args = ['--pyargs', 'skimage']
-        if verbose:
-            args.extend(['-v', '-s'])
-        if doctest:
-            args.extend(['--doctest-modules'])
-            # Make sure warnings do not break the doc tests
-            with warnings.catch_warnings():
-                warnings.simplefilter("ignore")
-                success = pytest.main(args)
-        else:
-            success = pytest.main(args)
-        # Return sys.exit code
-        if success:
-            return 0
-        else:
-            return 1
-
-
-# do not use `test` as function name as this leads to a recursion problem with
-# the nose test suite
-test = _test
-test_verbose = functools.partial(test, verbose=True)
-test_verbose.__doc__ = test.__doc__
-doctest = functools.partial(test, doctest=True)
-doctest.__doc__ = doctest.__doc__
-doctest_verbose = functools.partial(test, doctest=True, verbose=True)
-doctest_verbose.__doc__ = doctest.__doc__
-
-
-# Logic for checking for improper install and importing while in the source
-# tree when package has not been installed inplace.
-# Code adapted from scikit-learn's __check_build module.
-_INPLACE_MSG = """
-It appears that you are importing a local scikit-image source tree. For
-this, you need to have an inplace install. Maybe you are in the source
-directory and you need to try from another location."""
-
-_STANDARD_MSG = """
-Your install of scikit-image appears to be broken.
-Try re-installing the package following the instructions at:
-http://scikit-image.org/docs/stable/install.html """
-
-
-def _raise_build_error(e):
-    # Raise a comprehensible error
-    local_dir = osp.split(__file__)[0]
-    msg = _STANDARD_MSG
-    if local_dir == "skimage":
-        # Picking up the local install: this will work only if the
-        # install is an 'inplace build'
-        msg = _INPLACE_MSG
-    raise ImportError("""%s
-It seems that scikit-image has not been built correctly.
-%s""" % (e, msg))
-
-try:
-    # This variable is injected in the __builtins__ by the build
-    # process. It used to enable importing subpackages of skimage when
-    # the binaries are not built
-    __SKIMAGE_SETUP__
-except NameError:
-    __SKIMAGE_SETUP__ = False
-
-if __SKIMAGE_SETUP__:
-    sys.stderr.write('Partial import of skimage during the build process.\n')
-    # We are not importing the rest of the scikit during the build
-    # process, as it may not be compiled yet
-else:
-    try:
-        from ._shared import geometry
-        del geometry
-    except ImportError as e:
-        _raise_build_error(e)
-    from .util.dtype import *
-
-
-def lookfor(what):
-    """Do a keyword search on scikit-image docstrings.
-
-    Parameters
-    ----------
-    what : str
-        Words to look for.
-
-    """
-    import numpy as np
-    import sys
-    return np.lookfor(what, sys.modules[__name__])
-
-
-del warnings, functools, osp, imp, sys
+__version__ = '0.15.dev0'
+from .util.dtype import *
```

Comment:
NEVER import *

Reply:
I'm with you on that one, but that line of code was buried so deep in the past that it was impossible to find it.

https://github.com/scikit-image/scikit-image/blob/master/skimage/__init__.py#L178

Reply:
There remains the question on what is the expected functionality. I left that one as a "Good first issue" in Issue #3156